src/react/sebs_snacks/src/components/recipePage.jsx

```diff
@@ -0,0 +1,131 @@
+                                        <input 
+                                            id="newIngredient" 
+                                            type="text"
+                                            className="form-element" 
+                                            placholder="Ingredient" 
+                                            value={newIngredient} 
+                                            onChange={(e) => {
+                                                setNewIngredient(e.target.value);}} /> {/* Rerenders the page to display the character previously typed */}
+                                        <button onClick={() => {addIngredient(recipe, newIngredient); setNewIngredient('')}}>Add ingredient to list</button> {/*Adds ingredient to lists, and clears input text field, updating the state, forcing the component to rerender*/}
+                                    </div>
+
+                                    <h2 id="method">Instructions</h2>
+                                    <div className="recipeMethod">
+                                        <ol id="recipeInstructionList">
+                                        {recipe.instructionList.map(function(currentValue, index) {
+                                            return <li key={index}>{currentValue}</li>;
+                                            })}
+                                        </ol>
+                                        <h4>Add an Instruction:</h4>
+                                        <input 
+                                            id="newInstruction" 
+                                            type="text"
+                                            className="form-element" 
+                                            placholder="Instruction" 
+                                            value={newInstruction} 
+                                            onChange={(e) => {
+                                                setNewInstruction(e.target.value);}} />
+                                        <button onClick={() => {addInstruction(recipe, newInstruction); setNewInstruction('')}}>Add instruction to list</button>
+                                    </div>
+                                </div>
+                                <img id="recipePic" src={`/images/${recipe.recipePic}`} alt={recipe.alt}/>
+                            </div>
+                        </div>
+                    )
+                        : (
+                            <div>
+                                {console.log("Loading...")}
+                                <div className="recipeBox">
+                                    <p> Loading.... </p>
+                                </div>
+                            </div>
+                        )}
+            </body>
+        </div>
+    )
+}
+
+function addIngredient (recipe, newIngredient) { //function adds "newIngredient" to ingredient array in recipeData 
```

Comment:
Its a best practice to 
1. put your functions right below your imports. 
2. Use arrow notation when writing function headers.This is to keep things consistent because we sometimes write arrow functions inline, like onChange={(e) =>  setNewInstruction(e.target.value);}}